services/src/main/java/org/keycloak/services/managers/ApplianceBootstrap.java

```diff
@@ -127,4 +140,30 @@ public void createMasterRealmUser(String username, String password) {
         ServicesLogger.LOGGER.addUserSuccess(username, Config.getAdminRealm());
     }
 
+    public void createTemporaryMasterRealmAdminService(String clientId, String clientSecret, Integer expriationMinutes) {
+        RealmModel realm = session.realms().getRealmByName(Config.getAdminRealm());
+        session.getContext().setRealm(realm);
+
+        clientId = StringUtil.isBlank(clientId) ? DEFAULT_TEMP_ADMIN_SERVICE : clientId;
+        expriationMinutes = expriationMinutes == null ? DEFAULT_TEMP_ADMIN_EXPIRATION : expriationMinutes;
+
+        ClientModel adminClient = session.clients().addClient(realm, clientId);
+        adminClient.setEnabled(true);
+        adminClient.setServiceAccountsEnabled(true);
+        UserModel serviceAccount = session.users().addUser(realm, ServiceAccountConstants.SERVICE_ACCOUNT_USER_PREFIX + adminClient.getClientId());
```

Comment:
Do we really need to explicitly add the user like that? My feeling is this is not something we should do here directly. Maybe we could just call:
https://github.com/keycloak/keycloak/blob/64824bb77fff21af4e9c182a6a5bfdf9853fbb96/services/src/main/java/org/keycloak/services/managers/ClientManager.java#L156-L173

Reply:
It's there now. That was part of some missing work from yesterday - the patch I obtained from the old filesystem didn't have the last hour or so what I had done.